source/configuration/modules/mmkubernetes.rst

```diff
@@ -0,0 +1,357 @@
+
+.. csv-table::
+   :header: "type", "default", "mandatory", "|FmtObsoleteName| directive"
+   :widths: auto
+   :class: parameter-table
+
+   "word", "/etc/rsyslog.d/k8s_filename.rulebase", "no", "none"
+
+When processing json-file logs, this is the rulebase used to
+match the filename and extract metadata.  For the actual rules, see
+below `filenamerules`.
+
+.. _containerrules:
+
+containerrules
+^^^^^^^^^^^^^^
+
+.. csv-table::
+   :header: "type", "default", "mandatory", "|FmtObsoleteName| directive"
+   :widths: auto
+   :class: parameter-table
+
+   "word", "SEE BELOW", "no", "none"
+
+.. note::
+
+    This directive is not supported with liblognorm 2.0.2 and earlier.
+
+For journald logs, there must be a message property `CONTAINER_NAME`
+which has a value matching these rules specified by this parameter.
+The default value is::
+
+    rule=:%k8s_prefix:char-to:_%_%container_name:char-to:.%.%container_hash:char-to:\
+    _%_%pod_name:char-to:_%_%namespace_name:char-to:_%_%not_used_1:char-to:_%_%not_u\
+    sed_2:rest%
+    rule=:%k8s_prefix:char-to:_%_%container_name:char-to:_%_%pod_name:char-to:_%_%na\
+    mespace_name:char-to:_%_%not_used_1:char-to:_%_%not_used_2:rest%
+
+.. note::
+
+    In the above rules, the slashes ``\`` ending each line indicate
+    line wrapping - they are not part of the rule.
+
+There are two rules because the `container_hash` is optional.
+
+.. _containerrulebase:
+
+containerrulebase
+^^^^^^^^^^^^^^^^^
+
+.. csv-table::
+   :header: "type", "default", "mandatory", "|FmtObsoleteName| directive"
+   :widths: auto
+   :class: parameter-table
+
+   "word", "/etc/rsyslog.d/k8s_container_name.rulebase", "no", "none"
+
+When processing json-file logs, this is the rulebase used to
+match the CONTAINER_NAME property value and extract metadata.  For the
+actual rules, see `containerrules`.
+
+Example
+-------
+
+If you have a file `/etc/rsyslog.d/mmk8s.conf` which looks like this:
+
+.. code-block:: none
+
+    module(load="imfile" mode="inotify")
+    module(load="mmkubernetes" kubernetesurl="https://localhost:8443"
+           tls.cacert="/etc/rsyslog.d/mmk8s.ca.crt"
+           tokenfile="/etc/rsyslog.d/mmk8s.token" annotation_match=["."])
+
+    template(name="tpl" type="list") {
+             property(name="jsonmesg")
+             constant(value="\n")
+    }
+
+    ruleset(name="k8s") {
+            action(type="mmkubernetes")
+            action(type="omfile" file="/var/log/k8s.log" template="tpl")
+    }
+
+    input(type="imfile" file="/var/log/containers/*.log"
+          tag="kubernetes" addmetadata="on" ruleset="k8s")
+
+    if ($!_SYSTEMD_UNIT == "docker.service") and (strlen($!CONTAINER_NAME) > 0) then {
+        call k8s
+    }
+
+This assumes the files `/etc/rsyslog.d/k8s_filename.rulebase` and
+`/etc/rsyslog.d/k8s_container_name.rulebase` exist, as well as the
+files specified by `tls.cacert` and `tokenfile`.  This also assumes
+you have a `imjournal` input in order to use `CONTAINER_NAME`.
```

Comment:
Why not just add `imjournal` explicitly to the example and then move this line to the above revised introduction paragraph?